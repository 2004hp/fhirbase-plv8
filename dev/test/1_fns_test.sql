--db:fhirb -e
SET escape_string_warning=off;
--{{{
--test get_in_path
SELECT get_in_path('{"a": 1}'::jsonb, ARRAY['a']);
SELECT get_in_path('{"a": 1}'::jsonb, ARRAY['a', 'b']);
SELECT get_in_path('{"a":[{"b": [{"c":"c1"},{"c":"c2"}]}, {"b":[{"c":{"obj":"obj"}}]}]}'::jsonb, ARRAY['a','b','c']);
--}}}

--{{{
  SELECT is_array('[{"a":"b"}]'::jsonb);
--}}}

--{{{
select json_array_to_str_array(ARRAY['"a"'::jsonb,'"b"'::jsonb]);
--}}}
