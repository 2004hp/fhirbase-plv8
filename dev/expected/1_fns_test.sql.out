SET escape_string_warning=off;
SET
SELECT get_in_path('{"a": 1}'::jsonb, ARRAY['a']);
 get_in_path 
-------------
 {1}
(1 row)

SELECT get_in_path('{"a": 1}'::jsonb, ARRAY['a', 'b']);
 get_in_path 
-------------
 {}
(1 row)

SELECT get_in_path('{"a":[{"b": [{"c":"c1"},{"c":"c2"}]}, {"b":[{"c":{"obj":"obj"}}]}]}'::jsonb, ARRAY['a','b','c']);
               get_in_path                
------------------------------------------
 {"\"c1\"","\"c2\"","{\"obj\": \"obj\"}"}
(1 row)

SELECT is_array('[{"a":"b"}]'::jsonb);
 is_array 
----------
 t
(1 row)

select json_array_to_str_array(ARRAY['"a"'::jsonb,'"b"'::jsonb]);
 json_array_to_str_array 
-------------------------
 {a,b}
(1 row)

